File: compiler/bench/depth4.c
// expected: 500000000
// Stack depth 4 - first spill level, 500M ops
#include <stdio.h>

int main(void) {
    long stack[4];
    stack[0] = 0;  // acc
    stack[1] = 1;  // d1
    stack[2] = 2;  // d2
    stack[3] = 3;  // d3
    for (long i = 0; i < 500000000L; i++) {
        stack[0] = stack[0] + 1;
    }
    printf("%ld\n", stack[0]);
    return 0;
}

File: compiler/bench/depth4.fs
\ expected: 500000000
\ Stack depth 4 - first spill level, 500M ops
: main
  0 1 2 3                   ( acc d1 d2 d3 )
  500000000 0 do
    >r >r >r                ( acc ) ( R: d3 d2 d1 )
    1 +                     ( acc' )
    r> r> r>                ( acc' d1 d2 d3 )
  loop
  drop drop drop . cr
;
